33505: tr parents (#33927)
mdn/content: files/en-us/web/html/element/tr/index.md

## Usage notes

> -  Must be child of a {{HTMLElement("thead")}}, {{HTMLElement("tbody")}}, or {{HTMLElement("tfoot")}} element. The `<tr>` may be placed as a direct child of its parent {{HTMLElement("table")}} element, but only if the `<table>` has no child {{HTMLElement("tbody")}} elements, and only after any {{HTMLElement("caption")}}, {{HTMLElement("colgroup")}}, and {{HTMLElement("thead")}} elements. In this case, the `<tbody>` parent is implied.

Reviewer: It's not clear to me what the implications are of "the `<tbody>` is implied". Also I think this edit is confusing because the first sentence "Must be child of a {{HTMLElement("thead")}}, {{HTMLElement("tbody")}}, or {{HTMLElement("tfoot")}} element." implies that these are the only options, so it's confusing then to read the second sentence which presents another option. The previous version had "Otherwise" which helps with that.

Practically speaking AFAICS the thing to note is that browsers will actually add the `<tbody>` to the markup, so a selector like `table > tr` won't work, and this is surely confusing for people, and helpful to call out.

```suggestion
-  Must be child of a {{HTMLElement("thead")}}, {{HTMLElement("tbody")}}, or {{HTMLElement("tfoot")}} element. Additionally, the `<tr>` may be placed as a direct child of its parent {{HTMLElement("table")}} element, but only if the `<table>` has no child {{HTMLElement("tbody")}} elements, and only after any {{HTMLElement("caption")}}, {{HTMLElement("colgroup")}}, and {{HTMLElement("thead")}} elements. In this case, browsers will implicitly add the `<tbody>` to the generated markup. Note that this means selectors like `table > tr` will not select the `<tr>` element.
```

## Usage notes

- The `<tr>` element is valid as a child of a {{HTMLElement("thead")}}, {{HTMLElement("tbody")}}, or {{HTMLElement("tfoot")}} element only.

> - If the `<tr>` is placed as a direct child of its parent {{HTMLElement("table")}} element, the `<tbody>` parent is implied and browsers will actually add the `<tbody>` to the markup.

Reviewer 2: ```suggestion
- If the `<tr>` is placed as a direct child of its parent {{HTMLElement("table")}} element, the `<tbody>` parent is implied and browsers will add the `<tbody>` to the markup.
```

> - This implied `<tbody>` parent is only supported if the `<table>` otherwise has no child {{HTMLElement("tbody")}} elements, and only if the `<tr>` is included after any {{HTMLElement("caption")}}, {{HTMLElement("colgroup")}}, and {{HTMLElement("thead")}} elements.

Reviewer 2: ```suggestion
- The implied `<tbody>` parent is only supported if the `<table>` otherwise has no child `<tbody>` elements, and only if the `<tr>` is included after any {{HTMLElement("caption")}}, {{HTMLElement("colgroup")}}, and `<thead>` elements.
```
